Update NodeJS example with Query Service (#3322)
ydb-platform/ydb: ydb/docs/en/core/dev/example-app/_includes/example-nodejs.md

      logger.info('Making a simple typed select...');
      const result = await driver.queryClient.do({
          fn: async (session) => {
              const {resultSets} =
                  await session.execute({
                      rowMode: RowType.Ydb, // enables typedRows() on result sets
                      text: `
                          SELECT series_id,
                                 title,
                                 release_date
                          FROM ${SERIES_TABLE}
                          WHERE series_id = 1;`,
                  });
              const {value: resultSet1} = await resultSets.next();
              const rows: Series[] = [];
              // Note: resultSet1.rows will iterate YDB IValue structures
              for await (const row of resultSet1.typedRows(Series)) rows.push(row);
              return {cols: resultSet1.columns, rows};
          }
      });
      logger.info(`selectTypedSimple cols: ${JSON.stringify(result.cols, null, 2)}`);
      logger.info(`selectTypedSimple rows: ${JSON.stringify(result.rows, null, 2)}`);
  }
  ```

{% endlist %}

{% include [param_queries.md](steps/06_param_queries.md) %}

> There is no explicit Prepared Query option in the Query Service. YDB determines the necessity of using this mode by
> itself.

Reviewer: ```suggestion
```
We are talking about the query service in documentation only if it is absolutely necessary.

            for (const [seriesId, seasonId, episodeId] of data) {
                const episode = new Episode({seriesId, seasonId, episodeId, title: '', airDate: new Date()});
                const {resultSets, opFinished} = await session.execute({
                    parameters: {
                        '$seriesId': episode.getTypedValue('seriesId'),
                        '$seasonId': episode.getTypedValue('seasonId'),
                        '$episodeId': episode.getTypedValue('episodeId')
                    },
                    text: `
                        DECLARE $seriesId AS Uint64;
                        DECLARE $seasonId AS Uint64;
                        DECLARE $episodeId AS Uint64;

                        SELECT title,
                               air_date
                        FROM episodes
                        WHERE series_id = $seriesId
                          AND season_id = $seasonId
                          AND episode_id = $episodeId;`
                });
                const {value: resultSet} = await resultSets.next();
                const {value: row} = await resultSet.rows.next();
                await opFinished;

>                 logger.info(`Select prepared query ${JSON.stringify(row, null, 2)}`);

Reviewer: ```suggestion
                logger.info(`Parametrized select query ${JSON.stringify(row, null, 2)}`);
```

            }
        }
    });
}
```

{% include [scan-query.md](steps/08_scan_query.md) %}

> `QuerySession.execute()` method is used to retrieve data by a stream.

Reviewer: ```suggestion
The `QuerySession.execute()` method is used to retrieve data in a streaming manner.
```

```ts
async function selectWithParametrs(driver: Driver, data: ThreeIds[], logger: Logger): Promise<void> {

>     logger.info('Selecting prepared query...');

Reviewer: ```suggestion
    logger.info('Selecting with a parametrized query...');
```

            for (const [seriesId, seasonId, episodeId] of data) {

                const episode = new Episode({seriesId, seasonId, episodeId, title: '', airDate: new Date()});

                const {resultSets, opFinished} = await session.execute({
                    parameters: {
                        '$seriesId': episode.getTypedValue('seriesId'),
                        '$seasonId': episode.getTypedValue('seasonId'),
                        '$episodeId': episode.getTypedValue('episodeId')
                    },
                    text: `
                        DECLARE $seriesId AS Uint64;
                        DECLARE $seasonId AS Uint64;
                        DECLARE $episodeId AS Uint64;

                        SELECT title,
                               air_date
                        FROM episodes
                        WHERE series_id = $seriesId
                          AND season_id = $seasonId
                          AND episode_id = $episodeId;`
                });
                const {value: resultSet} = await resultSets.next();
                const {value: row} = await resultSet.rows.next();
                await opFinished;

>                 logger.info(`Select prepared query ${JSON.stringify(row, null, 2)}`);

Reviewer: ```suggestion
                logger.info(`Parametrized select query ${JSON.stringify(row, null, 2)}`);
```

            }
        }
    });
}
```

{% include [transaction-control.md](steps/10_transaction_control.md) %}

> Here's a code sample that demonstrates how to explicitly use the `Session.beginTransaction()`
> and `Session.сommitTransaction()` calls to create and terminate a transaction:

Reviewer: ```suggestion
Here's a code sample demonstrating how to explicitly use the `Session.beginTransaction()` and `Session.commitTransaction()` calls to execute a transaction:
```

{% list tabs %}

- do()

  ```ts
  async function explicitTcl(driver: Driver, ids: ThreeIds, logger: Logger) {

>       logger.info('Running prepared query with explicit transaction control...');

Reviewer: ```suggestion
      logger.info('Running a parametrized query with explicit transaction control...');
```

              const [seriesId, seasonId, episodeId] = ids;
              const episode = new Episode({seriesId, seasonId, episodeId, title: '', airDate: new Date()});
              await session.execute({
                  parameters: {
                      '$seriesId': episode.getTypedValue('seriesId'),
                      '$seasonId': episode.getTypedValue('seasonId'),
                      '$episodeId': episode.getTypedValue('episodeId')
                  },
                  text: `
                      DECLARE $seriesId AS Uint64;
                      DECLARE $seasonId AS Uint64;
                      DECLARE $episodeId AS Uint64;
 
                      UPDATE episodes
                      SET air_date = CurrentUtcDate()
                      WHERE series_id = $seriesId
                        AND season_id = $seasonId
                        AND episode_id = $episodeId;`
              })
              const txId = session.txId;
              await session.commitTransaction();
              logger.info(`TxId ${txId} committed.`);
          }
      });
  }
  ```

- doTx()

  ```ts
  async function transactionPerWholeDo(driver: Driver, ids: ThreeIds, logger: Logger) {

>       logger.info('Running query with one transaction per whole doTx()...');

Reviewer: ```suggestion
      logger.info('Running a query with one transaction per whole doTx()...');
```
